chain-abstraction/sol-implementation/edge_challenge_manager.go

```diff
@@ -62,14 +62,23 @@ func (e *specEdge) AssertionHash(_ context.Context) (protocol.AssertionHash, err
 }
 
 func (e *specEdge) TimeUnrivaled(ctx context.Context) (uint64, error) {
+	if e.timeUnrivaled.IsSome() {
```

Comment:
This is not a cachable field unless the edge is rivaled

Reply:
updated.